### PR TITLE
added new functionality to ios_command

### PR DESCRIPTION
**Pull request by privateip**
- commands argument now accepts a dict arguments[1]
- waitfor has been renamed to wait_for with an alias to waitfor
- only show commands are allowd when check mode is specified
- config mode is no longer allowed in the command stack
- add argument match with valid values any, all

[1] The commands argument will now accept a dict argument that can
specifiy the output format of the command.  To specify a dict argument
use the form of { command: <str>, output: <str>, prompt: <str>,
response: <str> }.  Command and output are required arguments. Output
accepts valid values text and json.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks again to @privateip. This module is going into community review.

We notice that you are one of the maintainers (@privateip, @gundalow) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: network/ios/ios_command.py [hunk 1]
```diff
@@ -155,29 +179,34 @@ def main():
...
+            if cmd['command'].startswith('conf'):
                 module.fail_json(msg='ios_command does not support running '
                                      'config mode commands.  Please use '
                                      'ios_config instead')
-            runner.add_command(cmd)
+            try:
+                runner.add_command(**cmd)
+            except AddCommandError:
+                exc = get_exception()
+                warnings.append('duplicate command detected: %s' % cmd)
```
Comment (gundalow) [inline network/ios/ios_command.py:203]: In a a different PR (Junos I believe) you silently accept if we end up running duplicated commands, but not here. Why is that?

File: network/ios/ios_command.py [hunk 2]
```diff
@@ -50,6 +50,17 @@
...
     version_added: "2.2"
+  match:
+    description:
+      - The I(match) argument is used in conjunction with the
+        I(wait_for) argument to specify the match policy.  Valid
+        values are C(all) or C(any).  If the value is set to C(all)
+        then all conditionals in the wait_for must be satisfied.  If
+        the value is set to C(any) then only one of the values must be
+        satisfied.
+    required: false
```
Comment (gundalow) [inline network/ios/ios_command.py:61]: Missing `choices`

**Comment by gundalow** [comment]
In the PR description you've written

> deprecated force argument, use match=none instead
> However there isn't a `match=none` (only 'any or 'all'), also no force option

Copy and paste from a previous PR, or should this be "WIP"?

needs_revision

File: network/ios/ios_command.py (hunk 1 above)
Comment (privateip) [inline network/ios/ios_command.py:203]: This is a change coming in 2.2 that will show up in all *_command modules

**Comment by gregdek** [comment]
Thanks @privateip for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by privateip** [comment]
commit msg updated.  Not sure what happened but its fixed now

**Comment by privateip** [comment]
ready_for_review

**Comment by gundalow** [comment]
shipit

**Pull request merged**